ow_lander/src/ow_lander/trajectory_sequence.py

```diff
@@ -83,11 +83,29 @@ def _append_trajectory(self, trajectory, planning_time):
       = list(self._get_final_joint_positions_of(trajectory))
     self._planning_time_total += planning_time
 
+  def _calculate_trajectory_distance(self, trajectory):
+    distance = 0.0
+    for i in range(len(trajectory.joint_trajectory.points) - 1):
+      point1 = trajectory.joint_trajectory.points[i].positions
+      point2 = trajectory.joint_trajectory.points[i + 1].positions
+      segment_distance = sum((x - y)**2 for x, y in zip(point1, point2)) ** 0.5
+      distance += segment_distance
+    return distance
+  
   def _plan(self):
     """Calls on MoveIt to plan the next trajectory of the sequence. If no
     trajectory is provided, the plan is constructed from
     """
-    success, trajectory, planning_time, error_code = self._group.plan()
+    best_distance = float('inf')
+    for i in range(5):
+      success_, trajectory_, planning_time_, error_code_ = self._group.plan()
+      distance = self._calculate_trajectory_distance(trajectory_)
```

Comment:
My Python isn't strong, but I'm not seeing what would make this call return possibly different values.  Everything looks deterministic and the same arguments are used all 5 times.  The variable `i` is just a counter.  I'm guessing that I'm missing something...

Reply:
Thank you for the reviews. Indeed it runs the same arguments all 5 times, however, each time it runs it will give a different answer. The .plan() function will generate a trajectory by using a planner from OMPL and Moveit! (For the arm default planner is RRTConnect). As I present on Dev day, the planner is probabilistic, which means every time we run the .plan() function, it might give us a different return trajectory.

Reply:
Ah I missed that inner call to plan() and so that's obviously the non-deterministic element here.  Thanks.